tests/aggregation/tests.py

```diff
@@ -1614,6 +1615,20 @@ def test_group_by_exists_annotation(self):
         ).annotate(total=Count("*"))
         self.assertEqual(dict(has_long_books_breakdown), {True: 2, False: 3})
 
+    def test_group_by_nested_expression_with_params(self):
+        books_qs = (
+            Book.objects.annotate(greatest_pages=Greatest("pages", Value(600)))
+            .values(
+                "greatest_pages",
+            )
+            .annotate(
+                min_pages=Min("pages"),
+                least=Least("min_pages", "greatest_pages"),
+            )
+            .values_list("least", flat=True)
+        )
+        self.assertCountEqual(books_qs, [300, 946, 1132])
```

Comment:
This also crashes on MySQL:
```sql
SELECT
    LEAST(MIN(`aggregation_book`.`pages`), GREATEST(`aggregation_book`.`pages`, %s)) AS `least`
FROM `aggregation_book`
GROUP BY GREATEST(`aggregation_book`.`pages`, %s) ORDER BY NULL
```
```
django.db.utils.OperationalError: (1055, "Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'test_django_2.aggregation_book.pages' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by")
```
but it's not a regression.

Reply:
This seems like a limitation of in MySQL's functional dependency detection logic that [must be worked around](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_any-value) using `ANY_VALUE` 🤢

The following seems to work

```sql
SELECT
    LEAST(MIN(`aggregation_book`.`pages`), GREATEST(ANY_VALUE(`aggregation_book`.`pages`), 600)) AS `least`
FROM `aggregation_book`
GROUP BY GREATEST(`aggregation_book`.`pages`, 600) ORDER BY NULL
```

[source](https://dbfiddle.uk/VYI6pdAq)

Thoughts about creating a dedicated MySQL issue for this? Do we know if Oracle is affected as well?

Reply:
> Do we know if Oracle is affected as well?

It works on Oracle.

Reply:
> Thoughts about creating a dedicated MySQL issue for this?

Added, see ticket-43262. I'll add a test skip on MySQL.

Reply:
More details on the MySQL the last paragraph of their `Handling of GROUP BY` [section of their docs](https://dev.mysql.com/doc/refman/8.0/en/group-by-handling.html)

It points at using a subquery mask to achieve the same result which is also a solution that would address psycopg3 server-side binding issues discussed on the ticket. The following query works just fine and is semantically equivalent

```sql
SELECT LEAST(min_pages, greatest_pages) AS `least` FROM (
  SELECT
    GREATEST(`aggregation_book`.`pages`, 600) greatest_pages,
    MIN(`aggregation_book`.`pages`) min_pages
  FROM `aggregation_book`
  GROUP BY 1 ORDER BY NULL
) masked
```

Reply:
> It points at using a subquery mask to achieve the same result which is also a solution that would address psycopg3 server-side binding issues discussed on the ticket. The following query works just fine and is semantically equivalent

I'm not very happy with the current solution (especially since it doesn't work on PostGIS :sob: ) so any alternatives are more than welcome.